packages/hardhat/contracts/Ethernauts.sol

```diff
@@ -12,6 +12,8 @@ contract Ethernauts is ERC721Enumerable, Ownable {
     using Address for address payable;
     using Strings for uint256;
 
+    error CustomError(string);
```

Comment:
Solidity CustomErrors are a gas-optimized way to define user-friendly errors that are self-explanatory.

In my opinion, we should abandon the current implementation that has the same semantic of a "normal" revert with something more self-explanatory like this:

`require(msg.value >= mintPrice, "Invalid msg.value")` should be replaced by 

```
error InsufficientPayment(uint256 sent, uint256 required);

[...contract code]

revert InsufficientPayment({
    sent: msg.value,
    required: mintPrice
});

```

There's another way to revert

```
revert InsufficientPayment(msg.value, mintPrice);
```

Reply:
this requires more effort because (like everything in code) we need to think which is the correct name for each error.

Some errors could be just defined like a GeneralError like you've done or IncorrectParameter

Reply:
I like this one! I mean if we introduce more code into the contract at least to be descriptive?

Reply:
So my idea was instead of adding a lot of multiple custom errors with different names, I could just define a single custom error that takes in a string and use that to replace all the require statement. This was done after I initially implemented it the way @StErMi mentioned and I felt that was a lot of code.

I think I still have the code to that implementation.

Reply:
@StErMi @mariobyn PR has been updated, what do you think?

Reply:
@geewynn it seems the CI doesn't pass due to some linting issues. https://github.com/ethernautdao/ethernauts/runs/4666538160?check_suite_focus=true

Reply:
Formatted with prettier, let's see if this will work.

Reply:
still failing.
https://github.com/ethernautdao/ethernauts/runs/4683403090?check_suite_focus=true
can you `run npm run lint` beforehand?

Reply:
I did run `npm run lint` and did not get this error.

I will check and try again